### PR TITLE
3.12.0 release notes

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: docs/about-mono/releases/3.12.0.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+---
+title: Release Notes Mono 3.12.0
+---
+
+Mono 3.12.0 is teh result of about 3 months of work, featuring over 600 commits and 68 fixes. It
```
Comment (akoeplinger) [inline docs/about-mono/releases/3.12.0.md]: `the` result

File: docs/about-mono/releases/3.12.0.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+Class Libraries
+---------------
+
+- OSX now uses an experimental FileSystemWatcher that relies on kqueue. It works and doesn't require polling.
+
+Known Issues
+============
+
+- The new OSX kqueue FileSystemWatcher will consume all file descriptors when watch very large directories.
```
Comment (akoeplinger) [inline docs/about-mono/releases/3.12.0.md]: `when watching`

File: docs/about-mono/releases/3.12.0.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+- Added support for UriComponents.SerializationInfoString. Fixes #21571.
+- Unify handling of default ctor and zero. Fixes #22591
+- Make SendAsync state free to handle recursive calls. Fixes #22724
+- Actually fix #22179 to allow concurrent requests.
+- DateTime.Parse now handles unlimited number of fractional digits. This change was made because .NET can parse ISO8601 format with an unlimited number of fractional digits. Fixes #22417.
+- Added test ISO8601FractionalDigits. Covers #22417.
+- Delay reflection hit to actual execution point. Workarounds #22522
+- Update RequestMessage.RequestUri.AbsoluteUri after redirect. Fixes #22383
+- Catch exceptions thrown by SerialPortStream.Dispose () when called from the finalizer. Fixes #15514.
+-KeyPairPresistence now tries to fix folder permissions. An exception is no longer thrown if the folder permissions can be fixed. Fixes bug 19274.
```
Comment (akoeplinger) [inline docs/about-mono/releases/3.12.0.md]: missing space after dash. also, it should be KeyPairP **er** sistence

Comment (Therzok) [inline docs/about-mono/releases/3.12.0.md]: @akoeplinger I prefer writing these remarks as ~~KeyPairPresistence~~ KeyPairPersistence.

Comment (Therzok) [inline docs/about-mono/releases/3.12.0.md]: Seems more readable to me, that's all.

File: docs/about-mono/releases/3.12.0.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+----------
+
+- Switched to adaptative hill climbing for sizing the pool. This fixes the cases where the old algorithm would cause thrashing by spinning too many threads.
+- The supervisor thread now goes into deep sleep when no new jobs are submitted for a while. This should save power on targets running on batteries.
+
+Runtime
+-------
+
+- Implement the new suspend machinery on windows, amd64 and arm. The new machinery is known to be more reliable and results in significantly less hangs
+when using advanced threading features of the platform.
```
Comment (Therzok) [inline docs/about-mono/releases/3.12.0.md]: Wrapping happened here for some reason?

File: docs/about-mono/releases/3.12.0.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+Performance
+-----------
+
+- X86 code now generate pushless code. This is faster on modern cpus and produces smaller unwind information.
+- Reworked internal runtime locking to do less work under hot locks and do more optimistic metadata loading.
+
+Class Libraries
+---------------
+
+- OSX now uses an experimental FileSystemWatcher that relies on kqueue. It works and doesn't require polling.
```
Comment (Therzok) [inline docs/about-mono/releases/3.12.0.md]: OS X

**Comment by kumpera** [comment]
@directhex is the windows installer blurb ok?

**Comment by akoeplinger** [comment]
@kumpera The new FileSystemWatcher isn't in 3.12 anymore, can you remove it from the notes?

**Pull request merged**